File: 프로그래머스/2/49994. 방문 길이/방문 길이.js
function solution(dirs) {
    let move = { L: [-1, 0], R: [1, 0], U: [0, 1], D: [0, -1] };
    let cur = [0, 0]
    let set = new Set()
    
    for(let dir of dirs){
        let nextX = cur[0] + move[dir][0]
        let nextY = cur[1] + move[dir][1]
        
        if(nextX >= -5 && nextX <= 5 && nextY >= -5 && nextY <= 5){
            set.add(`${cur[0]}${cur[1]}${nextX}${nextY}`);
            set.add(`${nextX}${nextY}${cur[0]}${cur[1]}`);
            cur = [nextX, nextY];
        }
    }
    return set.size / 2;
}

File: 프로그래머스/2/49994. 방문 길이/방문 길이.py
def solution(dirs):
    moves = {
        'L' : [-1, 0],
        'R' : [1, 0],
        'U' : [0 , 1],
        'D' : [0, -1]
    }
    pos = [0, 0]
    visited = set()
    for dir in dirs:
        nx = pos[0] + moves[dir][0]
        ny = pos[1] + moves[dir][1]
        if -5 <= nx <= 5 and -5 <= ny <= 5:
            visited.add(f'{pos[0]}{pos[1]}{nx}{ny}')
            visited.add(f'{nx}{ny}{pos[0]}{pos[1]}')
            pos = [nx, ny]
    return len(visited) // 2
    
